list/PageableList.js

```diff
@@ -315,8 +221,87 @@ define([
 			};
 		}),
 
+		/*jshint maxcomplexity:14*/
+		computeProperties: function (props) {
+			if (this.pageLength > 0) {
+				if ("_busy" in props || "hideOnPageLoad" in props || "autoPaging" in props || "showNoItems" in props) {
+					this._updateListView();
+				}
+
+				if ("loadPreviousMessage" in props || "loadingMessage" in props || "_showPreviousPageLoader" in props) {
+					if (this.previousPageLoader && this.previousPageLoader.deliver) {
+						this.previousPageLoader.labels = {
+							loadMessage: string.substitute(this.loadPreviousMessage, this),
+							loadingMessage: this.loadingMessage
+						};
+						this.previousPageLoader.deliver();
+					}
+				}
+
+				if ("loadNextMessage" in props || "loadingMessage" in props || "_showNextPageLoader" in props) {
+					if (this.nextPageLoader && this.nextPageLoader.deliver) {
+						this.nextPageLoader.labels = {
+							loadMessage: string.substitute(this.loadNextMessage, this),
+							loadingMessage: this.loadingMessage
+						};
+						this.nextPageLoader.deliver();
+					}
+				}
+			}
+		},
+
 		//////////// Private methods ///////////////////////////////////////
 
+		/*
+		 * Ovveride deliteful/List#_updateListview to update the widget view.
+		 * @private
+		 */
+		_updateListView: function () {
+			this._displayedPanel = (this._busy && this.hideOnPageLoad && !this.autoPaging) ? "loading-panel" :
+				(this.containerNode && this.containerNode.children.length > 0) ?
+					"list" : ((this.showNoItems) ? "no-items" : "none");
+		},
```

Comment:
Looking at this more, it really seems like code that should be in computeProperties().   (As you know) the only problem is how to trigger that code to run after `itemAdded()`, `itemRemoved()` etc. are called.  It seems like it would be cleaner for `itemAdded()` etc. to call `this.notifyCurrentValue("renderItems")`, and then for computeProperties to have code like:

```
if ("renderItems" in oldVals || "_busy" in oldVals || ... ) {
   this._displayedPanel = ...
}
```